src/main/java/io/fabric8/maven/docker/AbstractDockerMojo.java

```diff
@@ -425,6 +430,19 @@ protected void checkImageWithAutoPull(ServiceHub hub, String image, String regis
         }
     }
 
+    private synchronized Set<String> getPreviouslyPulledImageCache() {
+        @SuppressWarnings({ "rawtypes", "unchecked" })
+        Set<String> cache = (Set) session.getUserProperties().get(CONTEXT_KEY_PREVIOUSLY_PULLED);
+        if (cache == null) {
+            System.out.println("creating new context");
+
+            cache = Sets.newConcurrentHashSet();
```

Comment:
I could trigger an exception by calling the sample in data-jolokia. Comma
separated list sounds fine.

Jae Gangemi <notifications@github.com> schrieb am Di., 19. Juli 2016, 22:53:

> In src/main/java/io/fabric8/maven/docker/AbstractDockerMojo.java
> <https://github.com/fabric8io/docker-maven-plugin/pull/512#discussion_r71421049>
> :
>
> > @@ -425,6 +430,19 @@ protected void checkImageWithAutoPull(ServiceHub hub, String image, String regis
> >          }
> >      }
> >
> > +    private synchronized Set<String> getPreviouslyPulledImageCache() {
> > +        @SuppressWarnings({ "rawtypes", "unchecked" })
> > +        Set<String> cache = (Set) session.getUserProperties().get(CONTEXT_KEY_PREVIOUSLY_PULLED);
> > +        if (cache == null) {
> > +            System.out.println("creating new context");
> > +
> > +            cache = Sets.newConcurrentHashSet();
>
> i'm not sure what other sharing mechanisms we have and i'm going to guess
> that the plugin writes user properties and system properties right?
>
> i don't want to make this overly complicated...what about a comma
> delimited list?
>
> how do i trigger the integration tests locally?
>
> —
> You are receiving this because you modified the open/close state.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/fabric8io/docker-maven-plugin/pull/512/files/136bc012852fd2cba63e30beb2243ad18c2836ab#r71421049>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAGDCFK4qBMThQxxRNNX-xhm3W40ldIhks5qXTksgaJpZM4JJwOU>
> .
>